lib/Runtime/Library/DateImplementation.cpp

```diff
@@ -1073,7 +1073,22 @@ namespace Js {
         size_t size = sizeof(char16) * (ulength + 1);
         js_memcpy_s(pszSrc, size, psz, size);
 
+        #ifdef _WIN32
```

Comment:
Is this necessarily platform-dependent?

Reply:
Native windows _wcslwr_s is already fast - whilst this alternate path would work on windows it wouldn't be an optimisation there.

I'll look at putting it into PAL wcslwr as suggested by obastemur below - just will have to add some extra checks to only put it in when wanted.